lib/jct/term.rb

```diff
@@ -0,0 +1,56 @@
+module Jct
+  class Term
+    attr_reader :start_on, :end_on
+
+    def initialize(start_on:, end_on:)
+      raise ArgumentError.new('start_on data-type must be Date') unless start_on.is_a?(Date)
+      raise ArgumentError.new('end_on data-type must be Date') unless end_on.is_a?(Date)
+      raise ArgumentError.new('start_on must not be after than end_on') if start_on > end_on
+
+      @start_on = start_on
+      @end_on = end_on
+    end
+
+    def includes?(date)
+      start_on <= date && date <= end_on
+    end
+
+    def overlaps_with?(term)
+      # Patterns that self and term **donot** overlap
+      #   term.end_on < self.start_on
+      #     self:       |---|
+      #     term: |---|
+      #   OR
+      #   self.end_on < term.start_on
+      #     self:       |---|
+      #     term:             |---|
+      !(term.end_on < start_on || end_on < term.start_on)
+    end
+
+    def number_of_days
+      (end_on - start_on).to_i + 1
```

Comment:
Until now, the number of days has been calculated using `(start_on..end_on).count`. However, from the following simple benchmark, this method was faster than `(start_on..end_on).count`, so we changed it.

```ruby
# $ ruby -v
# ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [arm64-darwin21]

require 'benchmark'
require 'date'

n = 10000

d20200101 = Date.new(2020, 01, 01)
d20210101 = Date.new(2021, 01, 01)

use_count = Benchmark.realtime do
  n.times do |_|
    _count = (d20200101..d20210101).count
  end
end

unuse_count = Benchmark.realtime do
  n.times do |_|
    _count = (d20200101 - d20210101 + 1).to_i
  end
end

puts "use_count execution time: #{use_count}"
puts "unuse_count execution time: #{unuse_count}"

# => use_count execution time: 0.39297799998894334
# => unuse_count execution time: 0.0031369999051094055
```